packages/forms-web-app/src/dynamic-forms/middleware/get-journey-response-for-lpa.js

```diff
@@ -20,10 +21,12 @@ module.exports = () => async (req, res, next) => {
 	}
 
 	try {
-		const dbResponse = await getQuestionResponse(appealType, encodedReferenceId);
-		result = new JourneyResponse(appealType, referenceId, dbResponse?.answers, dbResponse.LPACode);
+		const dbResponse = await apiClient.getLPAQuestionnaire(referenceId);
+		const convertedResponse = mapDBResponseToJourneyResponseFormat(dbResponse);
+		result = new JourneyResponse(appealType, referenceId, convertedResponse, dbResponse.lpaCode);
 	} catch (err) {
 		logger.error(err);
+		await apiClient.postLPAQuestionnaire(referenceId, user.lpaCode);
```

Comment:
I don't understand this part, can you explain please?

Reply:
getLPAQuestionnaire will error if there is no questionnaire associated with that appeal case reference, so on catching the error this hits the post endpoint to create a new db entry (ie questionnaire) linked to that case ref.  In practice, would only expect this to be used when first looking to access a questionnaire from the dashboard (ie going to the task list).